sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -186,8 +212,7 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
             fileFormat = defaultSource,
             options = options)(sparkSession = sparkSession)
 
-          val created = LogicalRelation(relation,
-            catalogTable = Some(metastoreRelation.catalogTable))
+          val created = LogicalRelation(relation, catalogTable = Some(catalogTable))
```

Comment:
This would only be appropriate if the mode is INFER_AND_SAVE and the call to ```alterTable()``` succeeds, right? I'll see if I can refactor this method to do this without making it too convoluted.